### PR TITLE
rbac: remove unwanted rbac for leader election

### PR DESCRIPTION
**Pull request by humblec**
Endpoint  RBAC was removed via https://github.com/kubernetes-csi/external-resizer/commit/15a912cfe617c0d38a84cd2b8639c2013b912358 , however the comment was not adjusted which is
addressed in this PR.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

/kind cleanup

-->
```release-note
NONE
```

### CODE REVIEW

File: deploy/kubernetes/rbac.yaml [hunk 1]
```diff
@@ -27,21 +27,21 @@ rules:
...
-    resources: ["pods"]
-    verbs: ["get", "list", "watch"]
-  - apiGroups: [""]
-    resources: ["persistentvolumeclaims/status"]
-    verbs: ["patch"]
-  - apiGroups: [""]
-    resources: ["events"]
-    verbs: ["list", "watch", "create", "update", "patch"]
+  - apiGroups: [ "" ]
+    resources: [ "persistentvolumes" ]
```
Comment (pohly) [inline deploy/kubernetes/rbac.yaml:31]: Can you remove the unnecessary whitespace changes?

Comment (humblec) [reply deploy/kubernetes/rbac.yaml:31]: sure..

File: deploy/kubernetes/rbac.yaml [hunk 2]
```diff
@@ -58,20 +58,6 @@ roleRef:
   name: external-resizer-runner
   apiGroup: rbac.authorization.k8s.io
 
----
-# Resizer must be able to work with end point in current namespace
-# if (and only if) leadership election is enabled
```
Comment (pohly) [inline deploy/kubernetes/rbac.yaml:63]: This comment seems wrong. It should say "leases" instead of "end point".

Comment (pohly) [reply deploy/kubernetes/rbac.yaml:63]: https://github.com/kubernetes-csi/external-resizer/commit/15a912cfe617c0d38a84cd2b8639c2013b912358 removed the "endpoints" part without updating the comment.

File: deploy/kubernetes/rbac.yaml [hunk 3]
```diff
@@ -59,7 +59,8 @@ roleRef:
   apiGroup: rbac.authorization.k8s.io
 
 ---
-# Resizer must be able to work with end point in current namespace
+# Resizer must be able to work with leases based on the leader-election
+# setting.
 # if (and only if) leadership election is enabled
```
Comment (pohly) [inline deploy/kubernetes/rbac.yaml:64]: The full dot at the end of the line breaks the sentence. There's now a dangling part on the next line.

Comment (humblec) [reply deploy/kubernetes/rbac.yaml:63]: sure.. hopefully its correct now :)

**Review by pohly — APPROVED** [review]
/lgtm

nit: please also adjust the PR description.
